### PR TITLE
Add expenses to category for spend balance

### PR DESCRIPTION
**Pull request by nickzhihovich**
- Add models, controller, route for expense transactions
- Add views for creating expense transaction
- Add validation form for expense transaction
- Add creator service for expenses
- Add migration for expense transaction
- Add tests

download 17

### CODE REVIEW

File: spec/controllers/expense_transactions_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  describe 'POST #create' do
+    context 'when valid' do
+      let(:attributes) do
+        {
+          transaction: attributes_for(:transaction,
+            user: user,
+            expense_transactions_attributes: {
+              category_id: category.id
+            })
```
Comment (sinpai) [inline spec/controllers/expense_transactions_controller_spec.rb]: Seems like `)` should be at the same identation as `transaction`

File: spec/controllers/expense_transactions_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+    context 'when not valid' do
+      let(:attributes) do
+        {
+          transaction: attributes_for(:transaction,
+            expense_transactions_attributes: {
+              category_id: category.id
+            },
+            amount: nil)
```
Comment (sinpai) [inline spec/controllers/expense_transactions_controller_spec.rb]: the same

**Review by sinpai — COMMENTED** [review]
Suggestions

File: app/controllers/expense_transactions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+class ExpenseTransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :create_new_form, only: %i[new create]
+  before_action :set_categories, only: %i[new create]
```
Comment (AleksSenkou) [inline app/controllers/expense_transactions_controller.rb]: @marderer why not just:
```ruby
before_action :set_categories

def create
  if @form.validate(permitted_params)
    ExpenseTransactions::Creator.new(create_params).create
    redirect_to activity_page_path, flash: {notice: t('transaction_create')}
  else
    render :new
  end
end

def form
  @_form ||= ExpenseTransactionForm.new(
    current_user.transactions.new,
    expense_transactions: ExpenseTransaction.new
  )
end

```

**Review by AleksSenkou — COMMENTED** [review]
Looks good.

I'd like to leave a comment in expenses modal in order to understand what for I'm spending money

File: app/controllers/expense_transactions_controller.rb (hunk 3 above)
Comment (marderer) [reply app/controllers/expense_transactions_controller.rb]: @AleksSenkou I'm using `@form` for checks validation before creating

**Review by AleksSenkou — APPROVED** [review]
LGTM

**Pull request merged**